### DISCUSSION TITLE
Run discount functions when users log in to their account

### DISCUSSION THREAD

**Discussion by nboliver-ventureweb** [Shopify/function-examples#321] [Feedback: Discounts APIs]
Currently when a discount function relies on some piece of customer data (in our case metafields), if the customer builds a cart while logged in, the discount shows in the theme / liquid cart. If they proceed to log out, and then log back in, they will see the cart with no discounts applied - I guess because the function hasn't run again since they logged in?
It seems like it may be possible to work around this by triggering an empty cart update request with the ajax cart API after the user logs in, but this will require some theme wrangling which is inconvenient or impossible depending on the application of the function.
Ideally the customer should see their discounted cart immediately upon logging in, without needing to trigger an update or proceed to checkout.

**Comment by pablogiralt** [comment]
@nboliver-ventureweb any workarounds in the meantime? I understand the only way would be to trigger a cart update somehow, being modifying a cart attribute what comes to mind

**Reply by nboliver-ventureweb** [reply]
@pablogiralt From what I recall, I think an update request with no updates (empty object) will trigger the function to run
https://shopify.dev/docs/api/ajax/reference/cart#post-locale-cart-update-js

**Reply by pablogiralt** [reply]
will try, thanks!
